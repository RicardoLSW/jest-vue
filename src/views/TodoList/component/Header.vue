<template>
  <div class="header">
    <div>TodoList</div>
    <input
      data-test="input"
      v-model="inputValue"
      @keyup.enter="addTodoItem"
      placeholder="TodoItem"
    />
  </div>
</template>

<script>
export default {
  name: "Header",
  data() {
    return {
      inputValue: ""
    };
  },
  methods: {
    addTodoItem() {
      if (this.inputValue) {
        this.$emit("add", this.inputValue);
        this.inputValue = "";
      }
    }
  }
};
</script>

<style scoped lang="scss">
.header {
  height: 60px;
  background-color: #666;
  display: flex;
  align-items: center;
  justify-content: center;
  div {
    color: white;
    font-size: 24px;
    line-height: 60px;
  }
  input {
    width: 360px;
    margin-left: 50px;
    height: 30px;
  }
}
</style>
