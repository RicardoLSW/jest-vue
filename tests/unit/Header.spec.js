import { shallowMount } from "@vue/test-utils";
import Header from "@/views/TodoList/component/Header";

it("Header style snapshot", function() {
  const wrapper = shallowMount(Header);
  expect(wrapper).toMatchSnapshot();
});

it("Header should have input", function() {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  expect(input.exists()).toBe(true);
});

it("input value should null", function() {
  const wrapper = shallowMount(Header);
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe("");
});

it("inputValue should equal set value", function() {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue("ricardo luo");
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe("ricardo luo");
});

it("If inputValue has no value, the add event should not be triggered", function() {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue("");
  input.trigger("keyup.enter");
  expect(wrapper.emitted().add).toBeFalsy();
});

it("if inputValue has value, the add event should be triggered and clear inputValue", function() {
  const wrapper = shallowMount(Header);
  const input = wrapper.find('[data-test="input"]');
  input.setValue("ricardo luo");
  input.trigger("keyup.enter");
  expect(wrapper.emitted().add).toBeTruthy();
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe("");
});
