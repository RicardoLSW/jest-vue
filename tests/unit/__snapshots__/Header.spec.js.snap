// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header style snapshot 1`] = `
<div class="header">
  <div>TodoList</div> <input data-test="input" placeholder="TodoItem">
</div>
`;
